lib/game_logic.rb

```diff
@@ -0,0 +1,28 @@
+class Board
+  attr_reader :board
+  def initialize
+    @board = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    @count = 0
+  end
+
+  def update_board(player, option)
+    @count += 1
+    @board[option - 1] = player
+  end
+
+  def winner
+    board_patterns = [[@board[0], @board[1], @board[2]],
+                      [@board[3], @board[4], @board[5]],
+                      [@board[6], @board[7], @board[8]],
+                      [@board[0], @board[3], @board[6]],
+                      [@board[1], @board[4], @board[7]],
+                      [@board[2], @board[5], @board[8]],
+                      [@board[0], @board[4], @board[8]],
+                      [@board[2], @board[4], @board[6]]]
+    board_patterns.any?(%w[X X X]) || board_patterns.any?(%w[O O O])
+  end
```

Comment:
## Only one action

- The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Student's tends to create long methods that are doing multiple steps of the game process. (screenshot from Pathwright)
  - [ ] Please remove in another function

*Example:*

```ruby
 def winner_pattern
    board_patterns = [[@board[0], @board[1], @board[2]],
                      [@board[3], @board[4], @board[5]],
                      [@board[6], @board[7], @board[8]],
                      [@board[0], @board[3], @board[6]],
                      [@board[1], @board[4], @board[7]],
                      [@board[2], @board[5], @board[8]],
                      [@board[0], @board[4], @board[8]],
                      [@board[2], @board[4], @board[6]]]
end
```

Reply:
Thanks @bertil291utn. We have implemented this